### ISSUE TITLE
strange stacktrace when opening random test java files

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#672]
```
Java Model Exception: Java Model Status [resources/TestConfigGeneration/goldenoutput [in com.advantest.itee.tpl.tests] does not exist]
    at org.eclipse.jdt.internal.core.JavaElement.newNotPresentException(JavaElement.java:560)
    at org.eclipse.jdt.internal.core.PackageFragmentRoot.getUnderlyingResource(PackageFragmentRoot.java:719)
    at org.eclipse.jdt.internal.core.PackageFragment.getUnderlyingResource(PackageFragment.java:398)
    at org.eclipse.jdt.internal.core.Openable.getUnderlyingResource(Openable.java:335)
    at org.eclipse.jdt.internal.core.CompilationUnit.getUnderlyingResource(CompilationUnit.java:939)
    at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.getTraceToSource(TraceForTypeRootProvider.java:183)
    at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.createTraceToSource(TraceForTypeRootProvider.java:177)
    at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.getTraceToSource(TraceForTypeRootProvider.java:166)
    at org.eclipse.xtext.xbase.ui.editor.JavaEditorInputMatcher.matches(JavaEditorInputMatcher.java:80)
    at org.eclipse.ui.internal.WorkbenchPage.checkEditor(WorkbenchPage.java:2300)
    at org.eclipse.ui.internal.WorkbenchPage.findEditors(WorkbenchPage.java:2269)
    at org.eclipse.ui.internal.WorkbenchPage.busyOpenEditor(WorkbenchPage.java:3225)
    at org.eclipse.ui.internal.WorkbenchPage.access$26(WorkbenchPage.java:3190)
    at org.eclipse.ui.internal.WorkbenchPage$10.run(WorkbenchPage.java:3172)
    at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
    at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3167)
    at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3131)
    at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3121)
    at org.eclipse.ui.ide.IDE.openEditor(IDE.java:664)
    at org.eclipse.search.internal.ui.text.EditorOpener.showWithoutReuse(EditorOpener.java:73)
    at org.eclipse.search.internal.ui.text.EditorOpener.open(EditorOpener.java:44)
    at org.eclipse.search.ui.text.AbstractTextSearchViewPage.open(AbstractTextSearchViewPage.java:469)
    at org.eclipse.search.internal.ui.text.FileSearchPage.handleOpen(FileSearchPage.java:245)
    at org.eclipse.search.ui.text.AbstractTextSearchViewPage$4.open(AbstractTextSearchViewPage.java:763)
    at org.eclipse.ui.OpenAndLinkWithEditorHelper$InternalListener.open(OpenAndLinkWithEditorHelper.java:46)
    at org.eclipse.jface.viewers.StructuredViewer$2.run(StructuredViewer.java:851)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.ui.internal.JFaceUtil.lambda$0(JFaceUtil.java:44)
    at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
    at org.eclipse.jface.viewers.StructuredViewer.fireOpen(StructuredViewer.java:848)
    at org.eclipse.jface.viewers.StructuredViewer.handleOpen(StructuredViewer.java:1163)
    at org.eclipse.jface.util.OpenStrategy.fireOpenEvent(OpenStrategy.java:273)
    at org.eclipse.jface.util.OpenStrategy.access$2(OpenStrategy.java:268)
    at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:308)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
    at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5348)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1348)
    at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4602)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4183)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
    at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
    at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:681)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:595)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
    at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1472)
```

todo: check why this happens and if we can/should do something about that

**Comment by cdietrich** [comment]
trace with photon

```
9    [main] ERROR org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider  - Error finding trace to source
Java Model Exception: Java Model Status [nosrc/abc [in abc] does not exist]
	at org.eclipse.jdt.internal.core.JavaElement.newNotPresentException(JavaElement.java:570)
	at org.eclipse.jdt.internal.core.PackageFragmentRoot.getUnderlyingResource(PackageFragmentRoot.java:738)
	at org.eclipse.jdt.internal.core.PackageFragment.getUnderlyingResource(PackageFragment.java:419)
	at org.eclipse.jdt.internal.core.Openable.getUnderlyingResource(Openable.java:348)
	at org.eclipse.jdt.internal.core.CompilationUnit.getUnderlyingResource(CompilationUnit.java:980)
	at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.getTraceToSource(TraceForTypeRootProvider.java:184)
	at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.createTraceToSource(TraceForTypeRootProvider.java:178)
	at org.eclipse.xtext.common.types.ui.trace.TraceForTypeRootProvider.getTraceToSource(TraceForTypeRootProvider.java:167)
	at org.eclipse.xtext.xbase.ui.editor.JavaEditorInputMatcher.matches(JavaEditorInputMatcher.java:80)
	
```

[issue672.zip](https://github.com/eclipse/xtext-eclipse/files/2001838/issue672.zip)

**Comment by cdietrich** [comment]
TODO: is there a different possibility to get the ifile
if yes - use that
if not try to guard non open compilation units

**Comment by cdietrich** [comment]
Fixed in 2.14

**Comment by cdietrich** [comment]
thx @arshad1990
